### DISCUSSION TITLE
Is there any way to get sharp working with yarn PnP?

### DISCUSSION THREAD

**Discussion by cjroebuck** [yarnpkg/berry#6012] [Q&A] [answered]
Currently blocked on migrating to PnP because sharp [doesn't support yarn pnp](https://github.com/lovell/sharp/commit/004fff975ff5f3a933aebd15631fe12ed0c80035#commitcomment-134142091). 

I'm on an M1 mac, and node 20.9.0. This error happens on yarn 3.7.0 and yarn 4.0.2.

The actual install/build seems to work fine, and I can see the built files in the unplugged directory:

![CleanShot 2023-12-09 at 22 43 04@2x](https://github.com/yarnpkg/berry/assets/12208/53c55d6b-ef34-4e99-a808-fc818699d28b)

It is only at runtime when the error is thrown:

/Users/cjr/dev/.yarn/unplugged/sharp-npm-0.33.0-e95e460cd9/node_modules/sharp/lib/sharp.js:85
  throw new Error(help.join('\n'));
        ^

Error: Could not load the "sharp" module using the darwin-x64 runtime
ERR_DLOPEN_FAILED: dlopen(/Users/cjr/dev/.yarn/unplugged/@img-sharp-darwin-x64-npm-0.33.0-91623871fa/node_modules/@img/sharp-darwin-x64/lib/sharp-darwin-x64.node, 0x0001): Library not loaded: @rpath/libvips-cpp.42.dylib
  Referenced from: <856A1926-55E9-3FF2-B3C4-0C58C580F053> /Users/cjr/dev/.yarn/unplugged/@img-sharp-darwin-x64-npm-0.33.0-91623871fa/node_modules/@img/sharp-darwin-x64/lib/sharp-darwin-x64.node
  Reason: tried: '/Users/cjr/dev/.yarn/unplugged/@img-sharp-darwin-x64-npm-0.33.0-91623871fa/node_modules/@img/sharp-darwin-x64/lib/../../sharp-libvips-darwin-x64/lib/libvips-cpp.42.dylib' (no such file), '/Users/cjr/dev/.yarn/unplugged/@img-sharp-darwin-x64-npm-0.33.0-91623871fa/node_modules/@img/sharp-darwin-x64/lib/../../../sharp-libvips-darwin-x64/1.0.0/lib/libvips-cpp.42.dylib' (no such file), '/Users/cjr/dev/.yarn/unplugged/@img-sharp-darwin-x64-npm-0.33.0-91623871fa/node_modules/@img/sharp-darwin-x64/lib/../../node_modules/@img/sharp-libvips-darwin-x64/lib/libvips-cpp.42.dylib' (no such file), '/Users/cjr/dev/.yarn/unplugged/@img-sharp-darwin-x64-npm-0.33.0-91623871fa/node_modules/@img/sharp-darwin-x64/lib/../../../node_modules/@img/sharp-libvips-darwin-x64/lib/libvips-cpp.42.dylib' (no such file), '/Users/cjr/dev/.yarn/unplugged/@img-sharp-darwin-x64-npm-0.33.0-91623871fa/node_modules/@img/sharp-darwin-x64/lib/../../sharp-libvips-darwin-x64/lib/libvips-cpp.42.dylib' (no such file), '/Users/cjr/dev/.yarn/unplugged/@img-sharp-darwin-x64-npm-0.33.0-91623871fa/node_modules/@img/sharp-darwin-x64/lib/../../../sharp-libvips-darwin-x64/1.0.0/lib/libvips-cpp.42.dylib' (no such file), '/Users/cjr/dev/.yarn/unplugged/@img-sharp-darwin-x64-npm-0.33.0-91623871fa/node_modules/@img/sharp-darwin-x64/lib/../../node_modules/@img/sharp-libvips-darwin-x64/lib/libvips-cpp.42.dylib' (no such file), '/Users/cjr/dev/.yarn/unplugged/@img-sharp-darwin-x64-npm-0.33.0-91623871fa/node_modules/@img/sharp-darwin-x64/lib/../../../node_modules/@img/sharp-libvips-darwin-x64/lib/libvips-cpp.42.dylib' (no such file), '/usr/local/lib/libvips-cpp.42.dylib' (no such file), '/usr/lib/libvips-cpp.42.dylib' (no such file, not in dyld cache)

**Comment by lovell** [comment]
Please see https://github.com/lovell/sharp/issues/3888 for a possible future solution.

**Reply by lovell** [reply, marked as answer]
sharp v0.33.1 provides support for yarn PnP
